Zend/zend_API.c

```diff
@@ -2751,6 +2751,12 @@ ZEND_API int zend_set_hash_symbol(zval *symbol, const char *name, int name_lengt
 
 /* Disabled functions support */
 
+zend_op_array *display_disabled_compile_string(zval *source_string, char *filename)
+{
+	zend_error(E_WARNING, "eval() has been disabled for security reasons");
+	return NULL;
+}
+
 /* {{{ proto void display_disabled_function(void)
 Dummy function which displays an error when a disabled function is called. */
 ZEND_API ZEND_FUNCTION(display_disabled_function)
```

Comment:
The added `display_disabled_compile_string ` is because we can't use this one?

Reply:
Yes, the signatures are different. `display_disabled_function` has the signature of an internal function and it can be replaced as function pointer in the `function_table` (see `zend_disable_function` just below this code in the same file).

The `eval` disable support makes use of the function pointer to `zend_compile_string` which can be replaced with different implementation using the same idea as the function table disable functionality.